pkg/reconciler/shared/tektonconfig/tektonconfig.go

```diff
@@ -138,3 +154,45 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, tc *v1alpha1.TektonConfi
 
 	return nil
 }
+
+func (r *Reconciler) createOperatorVersionConfigMap(tc *v1alpha1.TektonConfig) error {
+	koDataDir := os.Getenv(common.KoEnvKey)
+	operatorDir := filepath.Join(koDataDir, "info")
+
+	if err := common.AppendManifest(&r.manifest, operatorDir); err != nil {
+		return err
+	}
+
+	manifest, err := r.manifest.Transform(
+		mf.InjectNamespace(tc.GetSpec().GetTargetNamespace()),
+	)
+	if err != nil {
+		return err
+	}
+
+	if err = manifest.Apply(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Reconciler) ensureTargetNamespaceExists(ctx context.Context, tc *v1alpha1.TektonConfig) error {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: tc.GetSpec().GetTargetNamespace(),
+			Labels: map[string]string{
+				"operator.tekton.dev/targetNamespace": tc.GetSpec().GetTargetNamespace(),
```

Comment:
what is the purpose of this label.

🧑‍💻 if it is to query kubernetes namespaces api and find the current tekton targetnamespace, then the `value` part should be something constant irrespective of the value of targetNamespace field in the CR.

eg: `"operator.tekton.dev/targetNamespace": true`

if the `value` part of this label could change, then this label becomes useless.

In addition, if we already know the name then why don't we just `get` the namespace using it as the metadata.name will already have the name.